sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/trace/_exporter.py

```diff
@@ -101,41 +101,49 @@ def _convert_span_to_envelope(span: Span) -> TelemetryItem:
                 envelope.tags["ai.cloud.role"] = service_name
         if service_instance_id:
             envelope.tags["ai.cloud.roleInstance"] = service_instance_id
-
+        else:
+            envelope.tags["ai.cloud.roleInstance"] = platform.node()  # hostname default
+        envelope.tags["ai.internal.nodeName"] = envelope.tags["ai.cloud.roleInstance"]
     envelope.tags["ai.operation.id"] = "{:032x}".format(span.context.trace_id)
-    parent = span.parent
-    if parent:
+    if "enduser.id" in span.attributes:
+        envelope.tags["ai.user.id"] = span.attributes["enduser.id"]
+    if span.parent:
         envelope.tags["ai.operation.parentId"] = "{:016x}".format(
-            parent.span_id
+            span.parent.span_id
         )
+    # pylint: disable=too-many-nested-blocks
     if span.kind in (SpanKind.CONSUMER, SpanKind.SERVER):
         envelope.name = "Microsoft.ApplicationInsights.Request"
         data = RequestData(
-            name=span.name,
+            name=span.name[:1024],  # Breeze max length
             id="{:016x}".format(span.context.span_id),
             duration=_utils.ns_to_duration(span.end_time - span.start_time),
-            response_code=str(span.status.status_code.value),
+            response_code="0",
             success=span.status.is_ok,
```

Comment:
Unset is also considered successful
https://github.com/microsoft/common-schema/blob/main/PartB/traces/Span.md#success

Reply:
`is_ok` returns true if status is `unset`. https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-api/src/opentelemetry/trace/status.py#L77